### PR TITLE
Simplify the findHashIndex() function

### PR DESCRIPTION
**Pull request by AndrejMitrovic**

### CODE REVIEW

File: source/agora/consensus/data/Block.d [hunk 1]
```diff
@@ -334,25 +335,11 @@ public struct Block
 
         const size_t size = min(this.txs.length, this.merkle_tree.length);
```
Comment (AndrejMitrovic) [inline source/agora/consensus/data/Block.d]: I am not sure why this constant is here though. The merkle tree's length should be `(tx.length * 2) - 1`, so I think this should be moved to the assert above as a check and this `min` should be removed.

**Comment by AndrejMitrovic** [comment]
Added a commit removing the `size` constant.

**Review by Geod24 — DISMISSED** [review]
The Merkle Tree is not sorted. The transactions are.

**Comment by AndrejMitrovic** [comment]
Well in the current implementation the merkle tree array is sorted from [0 .. txs.length], whereas the actual transaction array is not sorted at all.

**Comment by AndrejMitrovic** [comment]
Yeah, I've fixed the commit message now.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**